### ISSUE TITLE
max grpc message size is too small

### ISSUE THREAD

**Issue by greaber** [tensorflow/tensorboard#1202]
I am using TensorBoard 1.8.0.  I am trying to use the debugger, but I get this error message:

```
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with (StatusCode.RESOURCE_EXHAUSTED, Received message larger than max (6561349 vs. 4194304))>
```

Someone asked about this problem here: https://stackoverflow.com/questions/49648395/tensorboard-debugger-and-grpc-max-message-size-between-client-and-server

A comment on that question suggests that setting `grpc.max_receive_message_size` could help, but there is no variable named `max_receive_message_size` defined by the grpc module, and defining one doesn't do anything.  So I guess this limit needs to be set some other way?

I am surprised that I am hitting this issue since my model seems of ordinary size, so I would expect that I would not hit this issue unless lots of people were hitting it(?)

**Comment by jart** [comment]
This is probably a duplicate of #1103, most likely due to a large `tf.GraphDef`. cc: @caisq on debugger graph gRPC overflow.

**Comment by greaber** [comment]
Yep, this must be the issue since setting `send_traceback_and_source_code=False`, as suggested in #1103, makes the debugger work.

**Comment by caisq** [comment]
@greaber Thanks for the info. It is helpful. It shows that the root cause of the issue is sending of the traceback and source code data from TensorFlow runtime to the debugger backend is not sharded properly, causing failure in the case of large source code due to, say large source files or many source files. The GraphDef and tensor values are already sharded before sending.
I'll work on a fix to #1103 soon.
